src/cli/options/options.go

```diff
@@ -50,10 +51,40 @@ var (
 	skipReduceFV         bool
 )
 
+// s3 bucket info from flags
+var (
+	Bucket          string
+	Endpoint        string
+	Prefix          string
+	DoNotUseTLS     bool
+	DoNotVerifyTLS  bool
+	AccessKey       string
+	SecretAccessKey string
+	SessionToken    string
+)
+
 // AddGlobalOperationFlags adds the global operations flag set.
 func AddGlobalOperationFlags(cmd *cobra.Command) {
 	fs := cmd.PersistentFlags()
 	fs.BoolVar(&noStatsFV, NoStatsFN, false, "disable anonymous usage statistics gathering")
+
+	// Flags addition ordering should follow the order we want them to appear in help and docs:
+	// More generic and more frequently used flags take precedence.
+
+	// S3 flags
+	fs.StringVar(&Bucket, "bucket", "", "Name of S3 bucket for repo. (required)")
+	fs.StringVar(&Prefix, "prefix", "", "Repo prefix within bucket.")
+	fs.StringVar(&Endpoint, "endpoint", "s3.amazonaws.com", "S3 service endpoint.")
+	fs.BoolVar(&DoNotUseTLS, "disable-tls", false, "Disable TLS (HTTPS)")
+	fs.BoolVar(&DoNotVerifyTLS, "disable-tls-verification", false, "Disable TLS (HTTPS) certificate verification.")
+	fs.StringVar(&AccessKey, "access-key", "", "S3 access key")
+	fs.StringVar(&SecretAccessKey, "secret-access-key", "", "S3 access key")
+	fs.StringVar(&SessionToken, "session-token", "", "S3 session token")
+
+	// M365 flags
+	fs.StringVar(&credentials.AClientID, "client-id", "", "Azure app client ID")
```

Comment:
Sidestepping our current configuration communication patterns to have flags directly mutate global variables in lower level packages is an anti-pattern.